lib/molinillo/resolution.rb

```diff
@@ -134,7 +135,12 @@ def end_resolution
       # @return [void]
       def process_topmost_state
         if possibility
-          attempt_to_activate
+          begin
+            attempt_to_activate
+          rescue CircularDependencyError => e
+            create_conflict(e)
+            unwind_for_conflict until possibility && state.is_a?(DependencyState)
```

Comment:
Agreed. I wasn't sure of the best way to handle it. It seems like this logic might belong more inside `unwind_for_conflict` than here, but that method is used in a couple other places and I'm not quite familiar enough with the intricacies of its use to know if that was the right call or not.